### PR TITLE
[FEATURE] Update `DataAssistantResult.plot()` return value to emit `PlotResult` wrapper dataclass

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Add return value to `plot` (list of Altair charts) -> New `PlotResult` class
  - Note that these are the raw charts (no concatenation)
- Write tests to check for target strings in result objects

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py]: Should this be conditional? Should we always return the config?

I have concerns about polluting the notebook with config. Maybe a boolean flag in the signature?

Comment (cdkini) [reply great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py]: Additionally, we can use `chart.to_json()` as an alternative (returns a string). Thoughts here @NathanFarmer ?

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py]: We could have both `to_dict` and `to_json` available with boolean flag defaulted to `False`.

**Review by NathanFarmer — CHANGES_REQUESTED** [review]
I think we should default to no return value and allow boolean flags directly in `result.plot()` to allow for returning `dict` or json `str` representation. CC: @alexsherstinsky @donaldheppner

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py
Comment (donaldheppner) [reply great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py]: I'd rather not. Let's pick .to_dict() if we are going this way; it is no effort to convert a dictionary to json if folks want it that way. What is the use case for either (or both?)

**Review by donaldheppner — CHANGES_REQUESTED** [review]
I'm not sure why we are doing this ... and are we overburdening this method? Is it so we can create aggregate plot method results downstream? Couple of thoughts:
1) I'd rather return a stronger "Chart" type, even if it is a GE specific chart type, so folks know what the dictionary is intended to represent. I assume we went this way with dictionaries to avoid leaking implementation? We can instead solve the leak with encapsulation vs. dumping the raw data.
2) I do not want to manipulate the return type based on the input parameters (ie: returning either json or dict based on input parameters). This is where returning a richer type that produce the representation you want makes more sense. The richer type also allows you to encapsulate what you expect the caller to do with the result, so we're not having to duplicate that code across all callers.
3) for me, having a reference into a jira issue with ACs is helpful in supporting the review

File: great_expectations/rule_based_profiler/types/data_assistant_result/chart.py
Comment (alexsherstinsky) [inline great_expectations/rule_based_profiler/types/data_assistant_result/chart.py]: ❤️

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py
Comment (alexsherstinsky) [inline great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py]: ❤️

File: tests/test_utils.py
Comment (alexsherstinsky) [inline tests/test_utils.py:780]: @cdkini What can (or should?) we do about the following two questions:
1) We have `find_code_in_notebook()` and `replace_code_in_notebook()` in our `CLI` rendering utils.  Should these utilities be combined, or should they remain on their own?
2) When I was developing the above, if `find_strings_in_nested_obj()` existed and I had known about it, I would have attempted to leverage it.  What can we do towards making our team members aware of the useful utilities (for practical purposes as well as for avoiding re-implementing the same logic)?  Happy to brainstorm.  Thanks! /cc @donaldheppner

**Review by alexsherstinsky — APPROVED** [review]
LGTM (pending review/approval by @NathanFarmer ).  Thanks!

File: great_expectations/rule_based_profiler/types/data_assistant_result/chart.py
Comment (Shinnnyshinshin) [reply great_expectations/rule_based_profiler/types/data_assistant_result/chart.py]: :heart:

File: tests/test_utils.py
Comment (Shinnnyshinshin) [inline tests/test_utils.py:781]: recursion! :heart:

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you for this Chetan. LGTM pending Nathan and Don's comments.

File: tests/test_utils.py
Comment (cdkini) [reply tests/test_utils.py:780]: I think there might be a little bit of overlap but I'd be okay with keeping them separate for now. That feel like a very notebook-specific utility while this one is more general purpose.

And it's totally my fault for not being super vocal about this util! I think in large part it comes from just bringing it up during PR review ("hey I think this util would help your tests"). We should parse through our utils to see what's useful and when each one would be used.

Comment (cdkini) [reply tests/test_utils.py:780]: Maybe a README that documents the above?

File: great_expectations/rule_based_profiler/types/data_assistant_result/plot_result.py
Comment (donaldheppner) [inline great_expectations/rule_based_profiler/types/data_assistant_result/plot_result.py:19]: let's say we wanted to encapsulate our use of altair. Could we produce a property for "to_json" and "to_dict" and keep the chart itself private? </bad_idea>?

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py
Comment (donaldheppner) [inline great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py]: 👍

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py
Comment (donaldheppner) [inline great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py]: what if we replaced docs with spaces?

File: tests/test_utils.py
Comment (donaldheppner) [reply tests/test_utils.py:780]: re # 2, that is why I don't like utility methods, they are undiscoverable. :P

File: great_expectations/rule_based_profiler/types/data_assistant_result/plot_result.py
Comment (cdkini) [reply great_expectations/rule_based_profiler/types/data_assistant_result/plot_result.py:19]: hmm we definitely could but I think users are going to want to have the Altair Python objects since they can be directly updated during runtime. Some may just use the JSON output but I think we need to enable multiple workflows (even if that means exposing the library choice).

I think this is slightly different from our YAML situation in that this is recognized, well-supported library with a lot of recognition in the data space.

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py:100]: Necessary modification to make `get_line_chart` generic and work with both domains. I'm wary of bloating function signatures like this but this one simple change let's us reuse all the logic (since the only difference between the domains was a subtitle to reflect the column name).

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py:159]: Same note about generic methods.

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:175]: New helper that encapsulates all the shared logic between `_create_chart_for_table...` and `_create_chart_for_column`. This circumvents the need for code duplication but let me know if it feels bloated.

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:237]: Thought for a later PR - could we make these `namedtuples`? I think the pairing makes sense since they're always supplied together and it would cut down on the signature size.

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:237]: Sounds reasonable to me!

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:237]: Sounds reasonable to me!

**Review by NathanFarmer — APPROVED** [review]
LGTM!

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py
Comment (donaldheppner) [reply great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:237]: I think we could take it a step further and make these objects, encapsulating behavior to produce tool tips and titles for metrics and domains

**Review by donaldheppner — APPROVED** [review]

**Pull request merged**
